internal/queue/consumer.go

```diff
@@ -161,66 +160,95 @@ type WorkflowJob struct {
+	if j.CreatedAt == "" || j.CreatedAt == "null" {
+		return fmt.Errorf("%w: created_at in job_queued event", ErrMissingField)
+	}
+	createdAt, err := time.Parse(time.RFC3339, j.CreatedAt)
+	if err != nil {
+		return fmt.Errorf("%w: failed to parse created_at: %v", ErrInvalidTimestamp, err)
 	}
 
 	job := &database.Job{
 		ID:          strconv.Itoa(j.ID),
 		Platform:    platform,
 		Labels:      j.Labels,
-		CreatedAt:   *createdAt,
-		StartedAt:   parseToTime(j.StartedAt, c.logger),
-		CompletedAt: parseToTime(j.CompletedAt, c.logger),
+		CreatedAt:   createdAt,
+		StartedAt:   c.parseOptionalTime(j.StartedAt),
+		CompletedAt: c.parseOptionalTime(j.CompletedAt),
 		Raw:         c.extractRaw(j.Status, body),
 	}
 
 	if err := c.db.AddJob(ctx, job); err != nil {
 		if errors.Is(err, database.ErrExist) {
 			c.logger.Info("job already exists in database", "job_id", j.ID)
-			return NoRetryableError("job already exists in database")
+			// Job already exists is not retryable - it's expected in some cases
+			return fmt.Errorf("%w: job already exists in database", ErrNonRetryable)
 		}
-		return RetryableError(fmt.Sprintf("failed to add job to database: %v", err))
+		// Database errors are retryable
+		return fmt.Errorf("failed to add job to database: %w", err)
 	}
 
 	return nil
 }
 
+// parseOptionalTime parses an optional time string, returning nil if empty or invalid.
+func (c *AmqpConsumer) parseOptionalTime(timeStr string) *time.Time {
+	if timeStr == "" || timeStr == "null" {
+		return nil
+	}
+	t, err := time.Parse(time.RFC3339, timeStr)
+	if err != nil {
+		c.logger.Warn("failed to parse time", "timeStr", timeStr, "error", err)
+		return nil
+	}
+	return &t
+}
+
 // updateJobStartedInDB updates the job's started_at timestamp in the database.
 // If the job does not exist, it inserts the missing job to the database.
 func (c *AmqpConsumer) updateJobStartedInDB(ctx context.Context, j WorkflowJob, body []byte) error {
-	startedAt := parseToTime(j.StartedAt, c.logger)
-	if startedAt == nil {
-		return NoRetryableError("started_at missing in job_in_progress event")
+	if j.StartedAt == "" || j.StartedAt == "null" {
+		return fmt.Errorf("%w: started_at in job_in_progress event", ErrMissingField)
+	}
+	startedAt, err := time.Parse(time.RFC3339, j.StartedAt)
+	if err != nil {
+		return fmt.Errorf("%w: failed to parse started_at: %v", ErrInvalidTimestamp, err)
 	}
 
-	if err := c.db.UpdateJobStarted(ctx, platform, strconv.Itoa(j.ID), *startedAt, c.extractRaw(j.Status, body)); err != nil {
+	if err := c.db.UpdateJobStarted(ctx, platform, strconv.Itoa(j.ID), startedAt, c.extractRaw(j.Status, body)); err != nil {
 		if errors.Is(err, database.ErrNotExist) {
 			c.logger.Warn("job not found in database on start, inserting missing job", "job_id", j.ID)
 			return c.insertJobToDB(ctx, j, body)
 		}
-		return RetryableError(fmt.Sprintf("failed to update job started in database: %v", err))
+		// Database errors are retryable
```

Comment:
```suggestion
```
I thought the decision whether its retryable or not will be done in classifyError